src/test/java/com/google/sps/MatchServletTest.java

```diff
@@ -50,6 +53,7 @@ public void setup() {
         request = Mockito.mock(HttpServletRequest.class); 
         response = Mockito.mock(HttpServletResponse.class);
         sessionContext = Mockito.mock(SessionContext.class);
+        abuseFeature = Mockito.mock(AbuseDetection.class);
```

Comment:
You'll need to inject the 'abuseFeature' mock into matchServlet, similar to how the 'sessionContext' mock is also injected.
At present, the 'abuseFeature' mock you created isn't being used inside MatchServlet.

Reply:
Implemented this!